tests/contenttypes_tests/test_checks.py

```diff
@@ -216,3 +217,23 @@ class InvalidBookmark(models.Model):
                 id='fields.E001',
             )
         ])
+
+
+@isolate_apps('contenttypes_tests')
+class ModelCheckTests(SimpleTestCase):
+    def test_model_name_too_long(self):
+        class ModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModel(models.Model):
+            pass
+
+        errors = check_model_name_lengths()
+        self.assertEqual(1, len(errors))
+        self.assertEqual('contenttypes.E005', errors[0].id)
+        self.assertEqual(checks.ERROR, errors[0].level)
+        self.assertEqual('Model name is too long', errors[0].msg)
```

Comment:
I think it would be better to make a single check e.g.
```python
self.assertEqual(check_model_name_lengths(), [
    checks.Error(
        'Model name is too long.',
        hint='Model name should not be longer than 100 characters.',
        obj=ModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModel,
        id='contenttypes.E005',
    )
])
```